test/e2e/app-dir/actions/app-action.test.ts

```diff
@@ -1048,6 +1048,27 @@ createNextDescribe(
         expect(responseCodes).toEqual([303])
       })
 
+      it('merges cookies correctly when redirecting', async () => {
```

Comment:
I'm surprised that this test seems to be passing on canary without the changes in the PR, is there something more specific needed to trigger the buggy behavior?

Reply:
That's very strange - during a manual testing of `cookies().delete('foo')` it didn't work.

Reply:
@ztanner Okay I know what's happening! On an action redirection, `cookies().delete('foo')` only sets the value to `''` (an empty string), not actually deleting it.

Here's a reproduction:

https://github.com/vercel/next.js/assets/3676859/a7b8bf28-a8fe-4f3a-8ca6-33a47a4023ab

After clicking on the button, that /2 route should display `deleted` instead of `foo=`. The client cookie was correctly updated as it worked after refreshing though.